src/plugins/action_files_cleanup/src/cleanupFiles.ts

```diff
@@ -0,0 +1,93 @@
+import fs from 'fs'
+import path from 'path'
+import databaseMethods from './databaseMethods'
+
+const fileUploadPluginCode = 'fileUpload'
+
+interface FileInfo {
+  uniqueId: string
+  applicationResponseId: number
+  filePath: string
+  thumbnailPath: string
+}
+
+interface ResponseValue {
+  text?: string
+  files?: {
+    fileUrl: string
+    filename: string
+    mimetype: string
+    uniqueId: string
+    thumbnailUrl: string
+  }[]
+}
+
+module.exports['cleanupFiles'] = async function (input: any, DBConnect: any) {
+  const db = databaseMethods(DBConnect)
+
+  const { applicationData } = input
+  const applicationSerial = input?.applicationSerial || applicationData.applicationSerial
+  const applicationid = input?.applicationId || applicationData.applicationId
+  const {
+    environmentData: { appRootFolder, filesFolderName },
+  } = applicationData
+
+  console.log(`Processing files associated with Application ${applicationSerial}`)
+
+  try {
+    // Get all unsubmitted files for current application
+    const files = await db.getApplicationFiles(applicationSerial)
+
+    // Get responses that have uploaded file data
+    const fileResponses = (await db.getFileResponses(applicationid, fileUploadPluginCode))
+      .filter((response: ResponseValue) => response?.files)
+      .map((response: ResponseValue) => response.files)
+      .flat()
+
+    const responseFileUids = new Set(fileResponses.map((file: FileInfo) => file.uniqueId))
+
+    const deletedFiles: FileInfo[] = []
+    const submittedFiles: FileInfo[] = []
+
+    for (const file of files) {
+      if (!responseFileUids.has(file.uniqueId)) {
+        const resultFile = await deleteFile(file)
+        deletedFiles.push(resultFile)
+      } else {
+        await db.setFileSubmitted(file.uniqueId)
+        submittedFiles.push(file)
+      }
+    }
+    console.log('deletedFiles', deletedFiles)
+    return {
+      status: 'Success',
+      error_log: '',
+      output: {
+        deletedFiles,
+        submittedFiles,
+      },
+    }
+  } catch (error) {
+    console.log(error.message)
+    return {
+      status: 'Fail',
+      error_log: 'Problem cleaning up files: ' + error.message,
+    }
+  }
+
+  async function deleteFile(file: FileInfo) {
+    try {
+      // Delete the file
+      await fs.unlink(path.join(appRootFolder, filesFolderName, file.filePath), () => {})
```

Comment:
i think you want to promisify fs.unlink to make await work properly (not sure if it's awaiting for the unlink atm)

Reply:
Oh, it was my understanding that this was a proper async method. From what I can the fs module has Sync and Async versions of most of its functions. The sync version of this is called `unlinkSync` (I think).

Reply:
it's old school call back style, it would be something like this (i'll use timeout as example of operation)
```
const unlinkSync = (file, callback) => {
// do operation

setTimeout(() => {

callback('i am done')

}
 , operationTime)

}
```

promisify would wrap it in a promise:

```
(file) => new Promise(resolve => unlinkSync(file, resolve))
```

To test this you can specify a wrong file name and there won't be error thrown

Reply:
Cool, have used the fs.promises method now. d5d523c0